packages/financial-templates-lib/src/helpers/GasEstimator.js

```diff
@@ -66,21 +77,31 @@ class GasEstimator {
         message: "Gas estimator updated",
         networkId: this.networkId,
         lastUpdateTimestamp: this.lastUpdateTimestamp,
-        currentFastPriceGwei: this.lastFastPriceGwei,
+        currentMaxFeePerGas: this.latestMaxFeePerGasGwei,
+        currentMaxPriorityFeePerGas: this.latestMaxPriorityFeePerGasGwei,
+        lastFastPriceGwei: this.lastFastPriceGwei,
       });
     }
   }
 
-  // Returns the current fast gas price in Wei, converted from the stored Gwei value.
+  // Returns the current fast maxFeePerGas and maxPriorityFeePerGas.
   getCurrentFastPrice() {
-    // Sometimes the multiplication by 1e9 introduces some error into the resulting number,
-    // so we'll conservatively ceil the result before returning. This output is usually passed into
-    // a web3 contract call so it MUST be an integer.
-    return Math.ceil(this.lastFastPriceGwei * 1e9);
+    // Sometimes the multiplication by 1e9 introduces some error into the resulting number, so we'll conservatively ceil
+    // the result before returning.This output is usually passed into a web3 contract call so it MUST be an integer.
+    if (this.type == "london") {
+      return {
+        maxFeePerGas: Math.ceil(this.latestMaxFeePerGasGwei * 1e9),
+        maxPriorityFeePerGas: Math.ceil(this.latestMaxPriorityFeePerGasGwei * 1e9),
+      };
+    } else return { gasPrice: Math.ceil(this.lastFastPriceGwei * 1e9) };
```

Comment:
Out of curiosity, is there any _harm_ in including gasPrice when you also provide the base and priority fees? What does web3 default to in that case?

not suggesting a change -- purely curious

Reply:
this is a good point. I think we can include it without any _harm_. some reasons for excluding it could be:
1) We don't have a number for this so what would we put there? null perhaps?
2) having both `maxFeePerGas` & `maxPriorityFeePerGas` and `gasPrice` makes it a bit harder to reason about what kind of transaction this is; is it an London or legacy? well now rather than looking at the keys we need to look at keys + values and see if some are null
3) I'm not 100% sure that web3 will choose the right behavior. if you give it all 3 variables, with potentially `gasPrice` set to null or `maxFeePerGas` and `maxPriorityFeePerGas` will web3 consistently know to send either legacy or London? not really sure about this one so I'd rather be explicit